falcon-taz/src/main/java/pt/haslab/taz/TraceProcessor.java

```diff
@@ -264,7 +268,9 @@ private void parseJSONEvent( JSONObject event )
         switch ( type )
         {
             case LOG:
-                eventsPerThread.get( thread ).add( e );
+                String msg = event.getString( "message" );
+                LogEvent le = new LogEvent( e, msg );
```

Comment:
`le` is unfortunately a bad name for variables. Could it be more explicit, like `logEvent` ?
`msg` is a great name though, because it is understood by everyone.

The same for socket events.